### ISSUE TITLE
Use RespawnTemplates for JIP component

### ISSUE THREAD

**Issue by Wolfenswan** [ferstaberinde/F3#250]
Currently the JIP component works as follows:

```
1. Execute .sqf from init
2. sqf adds a "Respawn" eventhandler to player, which execVms init_onPlayerRespawn.sqf
3. init_onPlayerRespawn.sqf prompts JIP menu etc.
```

I assume this method dates back to ArmA or early ArmA2, before onPlayerRespawn.sqf was added by BIS, as _OnPlayerRespawn.sqf_ is funtionally identically to what we're doing with the first two steps mentioned above, i.e. our method is obsolete for quite a while now.
In ArmA3 the system has been improved further by providing cfgRespawnTemplate, allowing mission makers to define pre-made Respawn templates in the description.ext which can easily be replaced/disabled by changing a single line. We use this already for the F3 Spectator Component.

I suggest the following:

```
1. JIP component is defined as a respawn template:

// F3 - Respawn Templates
// Credits: Please see the F3 online manual (http://www.ferstaberinde.com/f3/en/)

class CfgRespawnTemplates
{
    // F3 Spectator Script
    class f_Spectator
    {
        onPlayerRespawn = "f_fnc_cam_init";
    };

    // F3 JIP Menu
    class f_JIP
    {
        onPlayerRespawn = "f\JIP\f_JIP_playerRespawn.sqf";
    };
};
2. By default, JIP is activated a the top:
respawnTemplates[] = {"f_spectator","f_JIP"};

To deactivate the JIP component a MM would only have to remove the "f_JIP" part.

3. In init.sqf F3 would provice global variables to control the way the JIP system works. For example:

f_var_JIP_FirstMenu = true;   // Do players connecting for the first time get the JIP menu?
f_var_JIP_RemoveCorpse = true; // Remove the old corpse of respawning players?
f_var_JIP_GearMenu = true;     // Can JIP/respawned players select their own gear? False will use gear assigned by F3 Gear Component if possible
```

In short: Using RespawnTemplates would remove redundancy due to obsolete method, would be robust (the old method could cause issues with eventhandlers) and "teach" about a ArmA functionality very relevant to mission makers interested in JIP/respawn (-> https://community.bistudio.com/wiki/Arma_3_Respawn).
